### PR TITLE
Issue 6064 - do not trigger a clash run if there are insufficient selections

### LINKED ISSUE #6064: Clash runs where there are no selection objects should be aborted straight away.
### Description
  Product issue: https://github.com/3drepo/3D-Repo-Product-Team/issues/639

  The clash run creation pipeline currently creates a run, writes a clash config, and queues `processClash` even if the resolved selections contain no composite object entries.

  This can happen if a plan has invalid/legacy stored selection data, or if selection rules resolve to no meshes/composite objects. In that case there is no useful clash work to process, but we still write a config file and push a task
  to `queueClashRun`.

  Update `backend/src/v5/processors/teamspaces/projects/clashes.js` so `createRun()` only writes the config file and queues the clash task after the selections have been resolved and confirmed to contain clashable composite objects.

  If no clashable objects are found, the run should be created but immediately marked as `aborted`, with the reason:

  `No clashable objects were found in the selected sets.`

  No clash config file should be written and no clash queue message should be sent in that case.

  ### Goals
  - [ ] As a project admin, when I start a clash run for a plan whose selections resolve to clashable objects, the run is queued and processed as normal.
  - [ ] As a project admin, when I start a clash run for a plan whose selections resolve to no clashable objects, the run is not queued for processing.
  - [ ] As a project admin, when a clash run has nothing to process, I can see that the run was aborted instead of remaining queued or failing later.
  - [ ] As a project admin, when a clash run is aborted because there are no objects to process, I can see a clear reason explaining that no clashable objects were found in the selected sets.

  ### Tasks
  - [ ] Refactor the config generation flow so selection resolution can be inspected before streaming/writing the final config.
  - [ ] Detect when resolved `selectionA`/`selectionB` produce no composite object entries.
  - [ ] Ensure `queueClashRun()` is only called after the resolved config has valid clashable object data.
  - [ ] Update the run status to `aborted` via `updateRunStatus()` when no clashable objects are found.
  - [ ] Ensure no clash config file is written when the run is aborted due to no clashable objects.
  - [ ] Add/update unit tests for `createRun()` covering valid selections still writing config and queueing successfully.
  - [ ] Add/update unit tests for `createRun()` covering selections that resolve to no objects.
  - [ ] Add/update unit tests for `createRun()` covering aborted empty runs not calling `queueClashRun()`.
  - [ ] Add/update route or integration coverage if needed to confirm the run response still returns the created run id.

### PR DESCRIPTION
**Pull request by carmenfan**
This fixes #6064

  #### Description

  - Abort clash runs before queueing when the configured selections do not produce any clash candidates.
  - Avoid writing a clash config file when there are no objects to clash test.
  - Batch bbox calculation for clash config generation by resolving bounds for groups of mesh nodes in one pass per container/revision.
  - Reduce repeated DB lookups by querying mesh nodes, parent transforms, and revision offset more efficiently.
  - Add UUID `unique()` helper for UUID-aware deduplication.
  - Update clash and scene processor coverage for aborted runs, grouped bbox calculation, missing mesh entries, and UUID helper behavior.

  #### Acceptance Criteria

  - [x] As a project admin running a clash test, I want a run with no resolved clash candidates to be marked as aborted immediately, so that an empty clash job is not queued.
  - [x] As a project admin running a clash test, I want no clash config file to be written when there are no objects to test, so that downstream processing is not triggered with invalid input.
  - [x] As a project admin running valid clash tests, I want bbox calculation to handle grouped object selections efficiently, so that large selections perform fewer repeated database queries.
  - [x] As a project admin running valid clash tests, I want self-intersection settings to still allow runs when only the configured self-intersection selection has objects.
  - [x] As a developer maintaining UUID-heavy code paths, I want shared UUID deduplication behavior, so that UUID strings and UUID objects are compared consistently.
  - [x] Existing clash run, scene bounds, and ticket comment behaviours remain covered by unit/e2e tests.

### CODE REVIEW

File: backend/src/v5/processors/teamspaces/projects/clashes.js [hunk 1]
```diff
@@ -147,13 +147,22 @@ const writeConfigSetEntry = async (teamspace, project, selections, stream, setNa
...
+				selectionEntries.push({
+					container,
+					revision,
+					objects: await applyExternalIds(teamspace, container, revision, objects),
+				});
+			}
+		}));
+
+	return selectionEntries;
+};
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply backend/src/v5/processors/teamspaces/projects/clashes.js:163]: We don't need it to be in order so this is fine.

File: backend/src/v5/processors/teamspaces/projects/clashes.js [hunk 2]
```diff
@@ -188,22 +209,56 @@ const writeConfigSetEntry = async (teamspace, project, selections, stream, setNa
...
+	const hasObjectsInA = !!selectionA.length;
+	const hasObjectsInB = !!selectionB.length;
+
+	if ((hasObjectsInA && hasObjectsInB)
+		|| (selfIntersectsA && hasObjectsInA)
+		|| (selfIntersectsB && hasObjectsInB)) {
+		await sendClashRunToQueue(teamspace, project, runId, context);
+	} else {
+		await updateRunStatus(teamspace, project, runId, clashRunStatus.ABORTED,
+			{ reason: 'The defined selections do not yield any candidates to execute a clash run.' });
```
Comment (ChristosTsiotsias) [inline backend/src/v5/processors/teamspaces/projects/clashes.js]: this has to be error : { reason }

Comment (ChristosTsiotsias) [reply backend/src/v5/processors/teamspaces/projects/clashes.js]: maybe we could handle that in the model method to ensure consistency

File: backend/src/v5/processors/teamspaces/projects/clashes.js [hunk 3]
```diff
@@ -188,22 +209,56 @@ const writeConfigSetEntry = async (teamspace, project, selections, stream, setNa
...
+	}
+
 	return runId;
```
Comment (ChristosTsiotsias) [inline backend/src/v5/processors/teamspaces/projects/clashes.js:264]: should we let the user know that the clash run will not run and no tickets will be created ? cause currently we rerturn the run Id as if it was successful

File: backend/tests/v5/e2e/routes/teamspaces/projects/clashes.test.js [hunk 4]
```diff
@@ -448,6 +455,24 @@ const testCreateRun = () => {
...
+			const res = await agent.post(route(teamspace, project.id,
+				planWithoutClashCandidates._id, users.tsAdmin.apiKey))
+				.expect(templates.ok.status);
+
+			const id = res.body._id;
+			const clashRun = await DB.findOne(teamspace, CLASH_RUNS_COL, { _id: stringToUUID(id) });
+			expect(clashRun).toEqual(expect.objectContaining({
+				status: clashRunStatus.ABORTED,
+				results: {
+					reason: 'The defined selections do not yield any candidates to execute a clash run.',
```
Comment (ChristosTsiotsias) [inline backend/tests/v5/e2e/routes/teamspaces/projects/clashes.test.js]: lets add the error here too

**Review by ChristosTsiotsias — CHANGES_REQUESTED** [review]

File: backend/src/v5/processors/teamspaces/projects/clashes.js (hunk 2 above)
Comment (carmenfan) [reply backend/src/v5/processors/teamspaces/projects/clashes.js]: I'd leave it for now - keen to just merge this in.

Besides this was done before I did the fixes for the previous PR

File: backend/src/v5/processors/teamspaces/projects/clashes.js (hunk 3 above)
Comment (carmenfan) [reply backend/src/v5/processors/teamspaces/projects/clashes.js:264]: it's the same deal as the user does not know when a clash run finishes.

don't worry about it for now - there are plans post release about email notifications and potentially UI showing the list of clash runs and their statuses.

**Review by ChristosTsiotsias — APPROVED** [review]
